File: Stack implementation through linked list in C, JS and Ruby/stack_through_linked_list.rb
# Stack implementation using Linked List in Ruby

# Node class for creating a node
class Node
  attr_accessor :value, :next

  def initialize(value)
    @value = value
    @next = nil
  end

end

# Linked List class for creating a linked list

class LinkedList

    def initialize
        @head = nil
    end

    # Push function to insert elements in the stack
    def add_first(value)
        new_node = Node.new(value)
        new_node.next = @head
        @head = new_node
        puts "Item pushed: #{value}"
    end

     # Pop function to remove elements from the stack
     def remove_first
        return nil if @head.nil?
        value = @head.value
        @head = @head.next 
        value
     end

     # isEmpty function returns true if the stack is empty
     def empty?
        @head.nil?
     end

     # Display function to display the elements of the linked list
     def display
        if empty?
            puts "Stack is empty"
            return
        end
        current = @head
        while current != nil
            puts current.value
            current = current.next
        end
     end

      # Clear function to clear the memory immediately
  def clear
    @head = nil
  end

end

# Stack class for creating a stack
class Stack 

    def initialize
        @linked_list = LinkedList.new
    end

    # Push function which calls the function of the LinkedList class
    def push(value)
        @linked_list.add_first(value)
    end

    # Pop function which calls the function of the LinkedList class
    def pop
    popped_value = @linked_list.remove_first
    if popped_value.nil?
        puts "Stack Underflow"
    else
        puts "Item popped: #{popped_value}"
        popped_value
    end
end

  # Show function which calls the display function of the LinkedList
  def show
    @linked_list.display
  end

  # Cleanup function which calls the clear function
  def cleanup
    @linked_list.clear
    puts "Memory cleaned up"
  end

end

# Main function to run the program
def main 
    stack = Stack.new
    choice = 0

    # Display the menu
    puts "\n*********Stack operations using linked list*********\n"
    puts "\n----------------------------------------------\n"

    # menu-driven program
    while choice != 4
        puts "\n\nChoose one from the below options...\n"
        puts "1. Push"
        puts "2. Pop"
        puts "3. Show"
        puts "4. Exit"
        print "Enter your choice: "
        choice = gets.chomp.to_i

        case choice
        when 1
            print "Enter the value: "
            value = gets.chomp.to_i
            stack.push(value)
        when 2
            stack.pop
        when 3
            stack.show
        when 4
            stack.cleanup
            puts "Exiting..."
        else
            puts "Invalid choice! Please enter a valid choice."
        end
    end

end

main()

File: Stack implementation through linked list in C, JS and Ruby/stack_through_linked_list.js
// Stack implementation using Linked List in JavaScript

// Node class for creating a node of the linked list
class Node {
    constructor(data) {
        this.value = data;
        this.next = null;
    }
}

// Linked List class and methods for stack operations
class LinkedList {
    constructor() {
        this.head = null;
    }

    // Push function to insert elements in the stack
    addFirst(value) {
        const newNode = new Node(value);
        newNode.next = this.head;
        this.head = newNode;
    }

    // Pop function to remove elements from the stack
    removeFirst() {
        if (!this.head) return null;
        const value = this.head.value;
        this.head = this.head.next;
        return value;
    }

    // isEmpty function retuns true if the stack is empty
    isEmpty() {
        return this.head === null;
    }

    // display function to display the Element of the Linked List
    display() {
        if(this.isEmpty()) {
            console.log("Stack is Empty");
            return;
        }
        let current = this.head;
        console.log("Printing Stack Elements");
        while(current) {
            console.log(current.value);
            current = current.next;
        }
        
    }

    // Clear function to clear the memory immediately
    clear() {
     this.head = null;       
    }
}

// Stack class and methods
class Stack {
    constructor() {
        this.linkedList = new LinkedList();
    }

    // Push function which calls the function of the Linked List class
    push(value) {
        this.linkedList.addFirst(value);
        console.log("Item Pushed")
    }

    // Pop function which calls the function of the Linked List class
    pop() {
        const poppedValue = this.linkedList.removeFirst();
        if (poppedValue === null) {
            console.log("Stack Underflow");
        }
        else {
            console.log("Item Popped");
            return poppedValue;
        }
    }

    // show function which calls the display function of the linked List
    show() {
        this.linkedList.display();
    }

    // cleanup function which calls the clear function
    cleanup() {
        this.linkedList.clear();
        console.log("Memory cleaned up");
    }
}

const stack = new Stack();
let choice = 0;

console.log("\n*********Stack operations using linked list*********\n");
console.log("\n----------------------------------------------\n");

const prompt = require("prompt-sync")();

// Menu Driven program 
while(choice !== 4) {
    console.log("\n\nChoose one from the below options...\n");
    console.log("1.Push\n2.Pop\n3.Show\n4.Exit");

    choice = parseInt(prompt("Enter your choice: "), 10);
    
    switch(choice) {
        case 1:
            const value = parseInt(prompt("Enter the value: "), 10);
            stack.push(value);
            break;
        case 2:
            stack.pop();
            break;
        case 3:
            stack.show();
            break;
        case 4:
            console.log("Exiting....");
            stack.cleanup();
            break;
        default:
            console.log("Please Enter a valid choice");
    }
}